src/Plugin/Field/FieldFormatter/MultiMedia.php

```diff
@@ -0,0 +1,435 @@
+      'video_formatter_width' => 0,
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+
+    $form['image_formatter'] = [
+      '#title' => $this->t('Image Formatter'),
+      '#type' => 'select',
+      '#options' => [
+        'responsive' => $this->t('Media Responsive Image Style '),
+        'static' => $this->t('Media Image Style'),
+        'rendered' => $this->t('Rendered entity'),
+      ],
+      '#default_value' => $this->getSetting('image_formatter'),
+    ];
+
+    $form['image_formatter_option'] = [
+      '#title' => $this->t('Image Formatter Setting'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('image_formatter_option'),
+      '#description' => $this->t('The machine name key for the display mode, style, etc.'),
+    ];
+
+    $form['video_formatter'] = [
+      '#title' => $this->t('Video Formatter'),
+      '#type' => 'select',
+      '#options' => [
+        'oembed' => $this->t('oEmbed Content'),
+        'rendered' => $this->t('Rendered entity'),
+      ],
+      '#default_value' => $this->getSetting('video_formatter'),
+    ];
+
+    $form['video_formatter_option'] = [
+      '#title' => $this->t('Video Formatter Setting'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_option'),
+      '#description' => $this->t('The machine name key for the display mode, style, etc.'),
+    ];
+
+    $form['video_formatter_height'] = [
+      '#title' => $this->t('Video Max Height (px)'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_height'),
+      '#description' => $this->t('Integer value for max height.'),
+    ];
+
+    $form['video_formatter_width'] = [
+      '#title' => $this->t('Video Max Width (px)'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_width'),
+      '#description' => $this->t('Integer value for max width.'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    foreach ($items as $delta => $item) {
+    // Get the media item.
+      $media_id = $item->getValue()['target_id'];
+      $media_item = Media::load($media_id);
+      $type = $media_item->bundle();
+
+      // Depending on the type of media referenced
+      switch ($type) {
+        case 'image':
+          $elements[$delta] = $this->viewImageElement($item, $media_item);
+          break;
+
+        case 'video':
+          $elements[$delta] = $this->viewVideoElement($item, $media_item);
+          break;
+
+        default:
+          $elements[$delta] = $this->viewDefaultElement($item, $media_item);
+      }
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Generates render arrays for the different image media type options.
+   *
+   * @param object $item
```

Comment:
what time of `object`?

Reply:
About half-past beer by the looks of things.